### PR TITLE
Add publish step to pipeline

### PR DESCRIPTION
**Pull request by AndreyTretyak**
Add publish step that will run on pushes to master

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -38,7 +38,7 @@
     <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
     <!-- The release version should get updated on every new release -->
     <Version>20.0.3</Version>
-    <PreReleaseSuffix Condition="'$(PreReleaseSuffix)' == ''">local</PreReleaseSuffix>
+    <PreReleaseSuffix Condition="'$(PreReleaseSuffix)' == ''">$([System.DateTime]::Now.ToString(yyyyMMdd-HHmmss))</PreReleaseSuffix>
```
Comment (muiriswoulfe) [inline Directory.Build.props:41]: `UtcNow` would be better in my view that `Now` as it is consistent irrespective of the timezone in use. It's also about 30 times more efficient

File: NuGet.Config [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (muiriswoulfe) [inline NuGet.Config:1]: Is this change deliberate?

File: Directory.Build.props (hunk 1 above)
Comment (muiriswoulfe) [inline Directory.Build.props:41]: I believe the "s" format would be good here and avoids using custom formatting. See https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings for more information.

If you add this, you should also add `CultureInfo.InvariantCulture` as the second parameter.

File: NuGet.Config (hunk 2 above)
Comment (AndreyTretyak) [reply NuGet.Config:1]: Yes, there was some invisible character at the beginning of the file

File: Directory.Build.props (hunk 1 above)
Comment (AndreyTretyak) [reply Directory.Build.props]: Good point, updated, thank you

Comment (AndreyTretyak) [reply Directory.Build.props]: This part would be used as version suffix for NuGet package (ex. 20.0.3-20200709-151556), so it might be sensible to `:` in "s" format (YYYY-MM-DDTHH:mm:ss), and in general, to me, it feels like in this case it's better to have direct control considering the usage of variable.

File: NuGet.Config (hunk 2 above)
Comment (muiriswoulfe) [reply NuGet.Config:1]: Makes sense. Probably the Unicode BOM.

File: Directory.Build.props (hunk 1 above)
Comment (muiriswoulfe) [reply Directory.Build.props]: Ah that makes sense, I agree skipping the `:` in the NuGet package version makes sense.

It might still be beneficial to add `CultureInfo.InvariantCulture` as the second parameter, as theoretically the serialisation can very per locale (although I don't believe it will for this format string in any current version of .NET).

Comment (AndreyTretyak) [reply Directory.Build.props]: I've tried a couple of options but I was not able to provide InvariantCulture into this MSBuild method call, so I'll probably leave it like this for now

Comment (muiriswoulfe) [reply Directory.Build.props]: Thanks for trying. My guess as to how this would work would be:
```
$([System.DateTime]::Now.ToString("yyyyMMdd-HHmmss", [System.Globalization.CultureInfo]::InvariantCulture))
```

If that doesn't work, should be fine as is.
